public/app/plugins/panel/geomap/migrations.ts

```diff
@@ -55,6 +71,47 @@ export function worldmapToGeomapOptions(angular: any): { fieldConfig: FieldConfi
     fieldConfig.defaults.decimals = v;
   }
 
+  // Set the markers range
+  const style = markersLayer.config!.style;
+  v = asNumber(angular.circleMaxSize);
+  if (v) {
+    style.size!.max = v;
+  }
+  v = asNumber(angular.circleMinSize);
+  if (v) {
+    style.size!.min = v;
+  }
+
+  let xform: DataTransformerConfig | undefined = undefined;
+  const reducer = fieldReducers.getIfExists(angular.valueName);
+  if (reducer && angular.locationData?.length) {
+    xform = {
+      id: DataTransformerID.reduce,
+      options: {
+        reducers: [reducer.id],
+      },
+    };
+
+    switch (angular.locationData) {
+      case 'countries':
+      case 'countries_3letter':
+        markersLayer.location = {
+          mode: FrameGeometrySourceMode.Lookup,
+          gazetteer: 'public/gazetteer/countries.json',
+          lookup: undefined, // will default to first string field from reducer
+        };
+        break;
```

Comment:
Will this handle full country name lookup the same as 3 letter?

Reply:
Yes -- https://github.com/grafana/grafana/blob/main/public/app/features/geo/gazetteer/worldmap.ts#L15 tries to match all the possible flavors

https://github.com/grafana/grafana/blob/main/public/gazetteer/countries.json